module/support/optimisation/NSGA2Optimiser/src/nsga2/Individual.cpp

```diff
@@ -0,0 +1,241 @@
+#include <fmt/format.h>
+#include <nuclear>
+
+namespace nsga2 {
+    Individual::Individual(const IndividualConfigurator& _config) : config(_config) {
+        id              = 0;
+        rank            = 0;
+        constrViolation = 0.0;
+        crowdDist       = 0.0;
+        evaluated       = false;
+        generation      = -1;
+        reals.clear();
+        gene.clear();
+        bins.clear();
+        objScore.clear();
+        constr.clear();
+
+        reals.resize(config.realVars, 0);
+        bins.resize(config.binVars, 0);
+        gene.resize(config.binVars);
+
+        if (int(config.binBits.size()) != config.binVars) {
+            NUClear::log<NUClear::WARN>(
+                fmt::format("binBits size ({}) != binVars ({})", config.binBits.size(), config.binVars));
+        }
+
+        for (int j = 0; j < config.binVars; j++) {
+            gene[j].resize(config.binBits[j], 0);
+        }
+        objScore.resize(config.objectives, 0);
+        constr.resize(config.constraints, 0);
+    }
+
+    void Individual::Initialize(const int& _id) {  // My entry point to init my parent pop to the one I want
+        id = _id;
+        NUClear::log<NUClear::INFO>("ID", id);
+
+        if (id == 0) {
+            // First individual gets real vars from config
+            // initialise real vars
+            for (int i = 0; i < config.realVars; i++) {
+                reals[i] = config.initialRealVars[i];
+            }
+        }
+        else {
+            // initialise real vars
+            for (int i = 0; i < config.realVars; i++) {
+                reals[i] = config.randGen->Real(config.realLimits[i].first, config.realLimits[i].second);
+            }
+        }
+        // initialise bin vars
+        for (int i = 0; i < config.binVars; i++) {
+            for (int j = 0; j < config.binBits[i]; j++) {
+                gene[i][j] = config.randGen->Realu() <= 0.5 ? 0 : 1;
+            }
+        }
+
+        for (auto r : reals) {
+            NUClear::log<NUClear::INFO>("Init Vars", r);
+        }
+    }
+
+    void Individual::Decode() {
+        int sum;
+        for (int i = 0; i < config.binVars; i++) {
+            sum = 0;
+            for (int j = 0; j < config.binBits[i]; j++) {
+                sum += (1 << (config.binBits[i] - 1 - j));
+            }
+
+            bins[i] = config.binLimits[i].first
+                      + (double) sum * (config.binLimits[i].second - config.binLimits[i].first)
+                            / (double) ((1 << (config.binBits[i])) - 1);
+        }
+    }
+
+    void Individual::CheckConstraints() {
+        if (config.constraints) {
+            constrViolation = 0.0;
+            for (int i = 0; i < config.constraints; i++) {
+                if (constr[i] < 0.0)
+                    constrViolation += constr[i];
+            }
+        }
+        else
+            constrViolation = 0.0;
+
+        evaluated = true;
+    }
+
+    // returns:  1 if this < _b (this dominates _b),
```

Comment:
Can this be proper docs comments?

Reply:
I will properly document. But I think this should stay. It makes it much easier to understand domination in the NSGA algorithm

Reply:
I don't think @LachlanCourt is saying it should be removed, I think he is saying it should be converted to doxygen style documentation directives https://nubook.nubots.net/guides/general/code-conventions#documentation